<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="Analysis of multi-sample Infinium MethylationEPIC data for the purpose of intratumor heterogeneity characterization.">
<title>Intratumor DNA Methylation Analysis â€¢ intratumormeth</title>
<script src="deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="deps/bootstrap-5.1.0/bootstrap.min.css" rel="stylesheet">
<script src="deps/bootstrap-5.1.0/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.rawgit.com/afeld/bootstrap-toc/v1.0.1/dist/bootstrap-toc.min.js"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="pkgdown.js"></script><meta property="og:title" content="Intratumor DNA Methylation Analysis">
<meta property="og:description" content="Analysis of multi-sample Infinium MethylationEPIC data for the purpose of intratumor heterogeneity characterization.">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light"><div class="container">
    
    <a class="navbar-brand me-2" href="index.html">intratumormeth</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.0.0.9000</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item">
  <a class="nav-link" href="reference/index.html">Reference</a>
</li>
<li class="nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <h6 class="dropdown-header" data-toc-skip>Analysis workflow</h6>
    <a class="dropdown-item" href="articles/preprocessing.html">Preprocessing</a>
    <a class="dropdown-item" href="articles/quality_control.html">Quality Control</a>
    <a class="dropdown-item" href="articles/copynumber.html">Copynumber</a>
  </div>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav"></ul>
</div>

    
  </div>
</nav><div class="container template-home">
<div class="row">
  <main id="main" class="col-md-9"><div class="section level1">
<div class="page-header"><h1 id="intratumormeth">intratumormeth<a class="anchor" aria-label="anchor" href="#intratumormeth"></a>
</h1></div>
<p>The goal of this package is to facilitate analysis of DNA methylation data obtained from Illumina MethylationEPIC platform for multiple samples of the same tumor.</p>
<div class="section level2">
<h2 id="installation">Installation<a class="anchor" aria-label="anchor" href="#installation"></a>
</h2>
<p>You can install the development version of intratumormeth from <a href="https://github.com/" class="external-link">GitHub</a> with:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># install.packages("devtools")</span>
<span class="fu">devtools</span><span class="fu">::</span><span class="fu"><a href="https://devtools.r-lib.org/reference/remote-reexports.html" class="external-link">install_github</a></span><span class="op">(</span><span class="st">"fynnwi/intratumormeth"</span><span class="op">)</span></code></pre></div>
</div>
<div class="section level2">
<h2 id="workflow">Workflow<a class="anchor" aria-label="anchor" href="#workflow"></a>
</h2>
<p>Every stage of the analysis pipeline is described in a separate vignette (see articles):</p>
<ol style="list-style-type: decimal">
<li>
<code>preprocessing</code>: read IDAT files, extract methylation values</li>
<li>
<code>quality_control</code>: investigate detection p-values, tumor purity, find poor quality samples and probes</li>
<li>
<code>copynumber</code>: run conumee and generate CNV data</li>
<li>
<code>classification</code>: evaluate glioma subtype and MGMT promoter methylation</li>
<li>
<code>dimensionality_reduction</code>: perform PCA and t-SNE to find clusters</li>
<li>
<code>phylogeny</code>: run TuMult to find phylogeny from CNV data</li>
</ol>
<!-- ## To be implemented: --><!-- Batch correction --><!-- - require user input of technical variables --><!--   - bisulfite conversion batch --><!--   - chip, row, column id --><!-- - run PCA and t-SNE and color by potential confounder variables --><!-- Probe filtering --><!-- - implement option to apply various filters to methylation data: --><!--   - sex chromosomes --><!-- Dimensionality reduction --><!-- - implement tsne function --><!-- Probe-wise heterogeneity analysis --><!-- - calculate intra-patient standard devation of beta values for each probe --><!-- - average these sds across all patients --><!-- - find probes that are commonly "methylation-instable"? --><!--   - what is their genomic context? distribution across chromosomes, regulatory features etc. --><!-- Gene-wise CNV analysis --><!-- - calculate intra-tumor mean and standard deviation for each gene (i.e. conumee detail region) --><!-- Sample table --><!-- | sample_id | patient_id | idat_location                                       | project | ceccarelli_subtype | capper_subtype  | mgmt_stp27 | --><!-- |-----------|------------|-----------------------------------------------------|---------|--------------------|-----------------|------------| --><!-- | sfb01_1   | sfb01      | ~/data/methylation_data/Pat01.1/202212330261_R07C01 | SFB824  | Classic-like       | RTK1            | M          | --><!-- | sfb01_2   | sfb01      | ~/data/methylation_data/Pat01.2/202242420132_R07C01 | SFB824  | PA-like            | GBM Mesenchymal | U          | --><!-- | ...       | ...        | ...                                                 | ...     | ...                | ...             | ...        | --><!-- Patient table --><!-- | patient_id | sex | age | n_samples | beta_mean | beta_sd | ... | --><!-- |------------|-----|-----|-----------|-----------|---------|-----| --><!-- | sfb01      | m   | 57  | 4         | 0.55      | 0.300   | ... | --><!-- | sfb02      | m   | 76  | 3         | 0.53      | 0.222   | ... | --><!-- | ...        | ... | ... | ...       | ...       | ...     | ... | --><!-- Detail region table --><!-- Collect cohort-level information about detail regions: --><!-- | detail_region | chr | start    | end      | n_probes | cnv_mean | cnv_sd | beta_mean | beta_sd | surroundings_cnv_mean | ... | --><!-- |---------------|-----|----------|----------|----------|----------|--------|-----------|---------|-----------------------|-----| --><!-- | PTEN          | 10  | 89621740 | 89725510 | 20       | -0.351   | 0.032  | 0.388     | 0.1     | 0.0234                | ... | --><!-- | PDGFRA        | ... | ...      | ...      | ...      | ...      | ...    | ...       | ...     | ...                   | ... | --><!-- | CDK2A/B       | ... | ...      | ...      | ...      | ...      | ...    | ...       | ...     | ...                   | ... | --><!-- | ...           | ... | ...      | ...      | ...      | ...      | ...    | ...       | ...     | ...                   | ... | --><!-- CpG table --><!-- Idea: summarize all cohort-level information related to single CpG sites in a table like this: --><!-- | probe_id     | mean | sd  | min | max | mean_intratumor_sd | sd_intratumor_mean | sd_intratumor_sd | neighborhood_correlation | --><!-- | ------------ | ---- | --- | --- | --- | ------------------ | ------------------ | ---------------- | ---- | --><!-- | cg12412432   | 1    |  1  | 0.1 | 0.9 | 0.002              | 0.75               | 0.0013           | ...  | --><!-- | cg12412432   | 1    |  1  | 0.1 | 0.9 | 0.002              | 0.75               | 0.0013           | ...  | --><!-- | ...          | ...  | ... | ... | ... | ...                | ...                | ...              | ...  | --><!-- ## Tumor purity --><!-- ### InfiniumPurify --><!-- - [paper](https://www.sciencedirect.com/science/article/pii/S2352304218300163) from 2018, 26 citations --><!-- - Wenger et al. use this tool --><!--   - using 51 adult brain tissue samples from GSE36278 (Sturm 2012, six normal brain samples), GSE40360 (Casaccia 2013, 19 control samples), GSE50798 (Dracheva 2013, postmortem 24 samples) and GSE52556 (Kleinman 2013, 34 control samples) as the normal data set --><!-- - normal tissue not necessarily required --><!--   - from vignette: "If either the number of tumor samples or number of normal smaples is less than 20, the tumor.type argument should be specified according to CancerTypeAbbr. If the numbers of tumor and normal samples are both more than 20, tumor.type could be null." --><!-- ### PAMES --><!-- - Purity Assesment from DNA MEthylation Sites --><!-- - [publication Benelli et al.](https://academic.oup.com/bioinformatics/article/34/10/1642/4792963?login=true) 2018 --><!--   - contains comparison to InfiniumPurify, they find high correlation --><!-- - [GitHub repo] (https://github.com/cgplab/PAMES) --><!-- - used by Verburg et al.  --><!--   - validation of purity estimates by looking at correlation of purity with chr7 amplification and chr10 deletion --><!-- - methylation data from normal tissue required! --><!-- Workflow: --><!-- 1. find CpG sites that are predictive for tumor purity by comparing tumor methylation against control group methylation data - these "informative CpG sites" are cancer type specific --><!-- 2. derive tumor purity (ratio of tumor to normal cells) from these informative sites --><!-- ## Continuous phylogeny tools --><!-- ### [BEAST](https://beast.community/index.html) --><!-- - there is also a [book from the author](https://www.amazon.com/dp/1107019656/) --><!-- - there exists [BEAST 2](http://www.beast2.org/) --><!--   - University of Auckland --><!-- ### [RevBayes](https://revbayes.github.io/) --><!-- - [publication](https://academic.oup.com/sysbio/article/65/4/726/1753608) with 360 citations --><!-- - developed by Sebastian HÃ¶hna (LMU Paleontology) --><!-- ### [Mesquite](http://www.mesquiteproject.org/) --><!-- - "modular system for evolutionary analysis" --><!-- - updated Aug 2021 --><!-- - support for continuous characters --><!-- ### [BayesPhylogenies](http://www.evolution.rdg.ac.uk/BayesPhy.html) --><!-- - University of Reading, UK --><!-- - from 2004 --><!-- ## Neuro-Oncology feedback --><!-- ### Reviewer 1 --><!-- - Ceccarelli classifier only discrete bins --><!-- - ~~correction for tumor purity~~ --><!-- - conclusions on copy number variations of individual genes are over interpreted --><!--   - many genes are on same chromosome -> LOH --><!-- - which CpGs make up most of the differences between biopsies?  --><!--   - Friedman test --><!-- - samples snap frozen or FFPE or mixed? --><!-- - evolutionary trajectories for all samples in supplementary information --><!-- ### Reviewer 2 --><!-- - clinical implications on heterogeneity? --><!-- - ~~tumor purity assessment based on methylation profiles~~ --><!-- - insufficient figure legends --><!-- - data availability missing --><!-- ### Reviewer 3 --><!-- - methylation heterogeneity missing --><!-- - ~~tumor purity~~ --><!--   - purity distribution as assessed by pathologist --><!-- - probe filtering (SNPs, sex chr etc.) --><!-- - settings for CNV analysis --><!-- - better description of TuMult --><!-- - classification probability scores --><!-- - percentage class-mes, mes-pa-like --><!-- - clustering:  --><!--   - color by Ceccarelli classification --><!--   - description of how patients clustered in relation to each other --><!-- - drop IDH-mut and use 914 CpGs described by Ceccarelli et al. to visualize samples and patients --><!-- - ~~tumor purity vs MGMT heterogeneity~~ --><!-- - compare MGMT heterogeneous and subtype heterogeneous patients --><!-- - Fig 4A y-axis label --><!-- - neutral vs. gain/loss heterogeneity clinically more interesting than medium vs. high gain/loss --><!-- - percentages how many samples (and patients) had specific CNV event --><!-- - phylogenetic distances between samples --><!-- - methylation vs. CNV heterogeneity --><!--   - do subtypes have their own branches in phylogeny? --><!-- - phylotrees of other patients in supplementary materials --><!-- - figure legends too brief --><!-- - compare CNV event frequencies with literature (Sottoriva, Taylor) -->
</div>
</div>
  </main><aside class="col-md-3"><div class="license">
<h2 data-toc-skip>License</h2>
<ul class="list-unstyled">
<li><a href="LICENSE.html">Full license</a></li>
<li><small>GPL (&gt;= 3)</small></li>
</ul>
</div>


<div class="citation">
<h2 data-toc-skip>Citation</h2>
<ul class="list-unstyled">
<li><a href="authors.html#citation">Citing intratumormeth</a></li>
</ul>
</div>

<div class="developers">
<h2 data-toc-skip>Developers</h2>
<ul class="list-unstyled">
<li>Fynn Withake <br><small class="roles"> Author, maintainer </small> <a href="https://orcid.org/0000-0002-3011-2463" target="orcid.widget" aria-label="ORCID" class="external-link"><span class="fab fa-orcid orcid" aria-hidden="true"></span></a> </li>
</ul>
</div>



  </aside>
</div>


    <footer><div class="pkgdown-footer-left">
  <p></p>
<p>Developed by Fynn Withake.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.2.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
