<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="intratumormeth">
<title>Copynumber • intratumormeth</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.1.0/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.1.0/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.rawgit.com/afeld/bootstrap-toc/v1.0.1/dist/bootstrap-toc.min.js"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Copynumber">
<meta property="og:description" content="intratumormeth">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">intratumormeth</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.0.0.9000</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="active nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <h6 class="dropdown-header" data-toc-skip>Analysis workflow</h6>
    <a class="dropdown-item" href="../articles/preprocessing.html">Preprocessing</a>
    <a class="dropdown-item" href="../articles/quality_control.html">Quality Control</a>
    <a class="dropdown-item" href="../articles/copynumber.html">Copynumber</a>
  </div>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav"></ul>
</div>

    
  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main"><div class="page-header">
      <img src="" class="logo" alt=""><h1>Copynumber</h1>
            
      
      
      <div class="d-none name"><code>copynumber.Rmd</code></div>
    </div>

    
    
<p>In this vignette, the package conumee <span class="citation">(Hovestadt and Zapatka 2017)</span> will be used to derive copy number profiles from the methylation intensity data. The resulting data will be further post-processed to facilitate downstream analysis.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">intratumormeth</span><span class="op">)</span>
<span class="co"># paths</span>
<span class="va">outputDir</span> <span class="op">&lt;-</span> <span class="st">"~/sfb824/packagepdgfra_output"</span>
<span class="va">inputDir</span> <span class="op">&lt;-</span> <span class="st">"~/sfb824/packagepdgfra_input"</span></code></pre></div>
<div class="section level2">
<h2 id="required-data">Required data<a class="anchor" aria-label="anchor" href="#required-data"></a>
</h2>
<p>In addition to sample methylation data, conumee requires methylation data of control samples with a flat genome. This reference data was kindly provided by Dr. Martin Sill from DKFZ. Furthermore, an <em>annotation object</em> has to be passed to conumee which tells the package which genomic regions should be evaluated. By means of the annotation object, a number of detail regions can be specified for which dedicated copy number values will be computed. For this study, all gene loci that are covered by at least 20 CpG probes on the EPIC array will be specified as detail regions.</p>
<p>Since different annotation and reference objects will be used for male and female samples, information about the patient sex is required. This is contained in <code>patientsheet</code>.</p>
<p>Load samplesheet and patientsheet:</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">samplesheet</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html" class="external-link">read.csv</a></span><span class="op">(</span><span class="st">"~/sfb824/packagepdgfra_input/samplesheet.csv"</span><span class="op">)</span>
<span class="va">samplesheet</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span> <span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="op">)</span>
<span class="co">#&gt;   sample_id patient_id                                       idat_location</span>
<span class="co">#&gt; 1   sfb01_1      sfb01 ~/data/methylation_data/Pat01.1/202212330261_R07C01</span>
<span class="co">#&gt; 2   sfb01_2      sfb01 ~/data/methylation_data/Pat01.2/202242420132_R07C01</span>
<span class="co">#&gt; 3   sfb01_3      sfb01 ~/data/methylation_data/Pat01.3/202212330261_R06C01</span>
<span class="co">#&gt; 4   sfb02_1      sfb02 ~/data/methylation_data/Pat02.1/202242420132_R02C01</span>
<span class="co">#&gt; 5   sfb02_2      sfb02 ~/data/methylation_data/Pat02.2/202242420132_R06C01</span>
<span class="co">#&gt; 6   sfb02_3      sfb02 ~/data/methylation_data/Pat02.3/202242420132_R01C01</span>
<span class="co">#&gt;   project</span>
<span class="co">#&gt; 1  SFB824</span>
<span class="co">#&gt; 2  SFB824</span>
<span class="co">#&gt; 3  SFB824</span>
<span class="co">#&gt; 4  SFB824</span>
<span class="co">#&gt; 5  SFB824</span>
<span class="co">#&gt; 6  SFB824</span>
<span class="va">patientsheet</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html" class="external-link">read.csv</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="va">inputDir</span>, <span class="st">"patientsheet.csv"</span><span class="op">)</span><span class="op">)</span>
<span class="va">patientsheet</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span> <span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="op">)</span>
<span class="co">#&gt;   patient_id n_samples sex age</span>
<span class="co">#&gt; 1 GU-HGG-154         4   m  NA</span>
<span class="co">#&gt; 2 GU-HGG-182         3   m  NA</span>
<span class="co">#&gt; 3 GU-HGG-185         3   m  NA</span>
<span class="co">#&gt; 4 GU-HGG-200         3   f  NA</span>
<span class="co">#&gt; 5 GU-HGG-204         3   f  NA</span>
<span class="co">#&gt; 6 GU-HGG-216         3   f  NA</span></code></pre></div>
</div>
<div class="section level2">
<h2 id="prepare-a-custom-conumee-annotation">Prepare a custom conumee annotation<a class="anchor" aria-label="anchor" href="#prepare-a-custom-conumee-annotation"></a>
</h2>
<p>The conumee annotation object can be generated by specifying start and stop coordinates of genomic regions for which a detailed output is desired. The start and stop coordinates of these detail regions will be extracted from the EPIC probes annotation.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/hansenlab/minfi" class="external-link">minfi</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">conumee</span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://bitbucket.com/kasperdanielhansen/Illumina_EPIC" class="external-link">IlluminaHumanMethylationEPICanno.ilm10b4.hg19</a></span><span class="op">)</span>
<span class="va">annoEpic</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/minfi/man/getAnnotation.html" class="external-link">getAnnotation</a></span><span class="op">(</span><span class="va">IlluminaHumanMethylationEPICanno.ilm10b4.hg19</span><span class="op">)</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span> 
  <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/as.data.frame.html" class="external-link">as.data.frame</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<p>This codeblock groups the probes by their associated genes and extracts positions of first and last probe for a given gene.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">genesEpic</span> <span class="op">&lt;-</span> <span class="va">annoEpic</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span> 
  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">UCSC_RefGene_Name</span> <span class="op">!=</span> <span class="st">""</span><span class="op">)</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span> 
  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span>gene <span class="op">=</span> <span class="fu">stringr</span><span class="fu">::</span><span class="fu"><a href="https://stringr.tidyverse.org/reference/str_split.html" class="external-link">str_split</a></span><span class="op">(</span><span class="va">UCSC_RefGene_Name</span>, <span class="st">";"</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span> 
  <span class="fu">tidyr</span><span class="fu">::</span><span class="fu"><a href="https://tidyr.tidyverse.org/reference/nest.html" class="external-link">unnest</a></span><span class="op">(</span>cols <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="va">gene</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span> 
  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/group_by.html" class="external-link">group_by</a></span><span class="op">(</span><span class="va">chr</span>, <span class="va">gene</span><span class="op">)</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span> 
  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/summarise.html" class="external-link">summarise</a></span><span class="op">(</span>from <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html" class="external-link">min</a></span><span class="op">(</span><span class="va">pos</span><span class="op">)</span>,
                   to <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html" class="external-link">max</a></span><span class="op">(</span><span class="va">pos</span><span class="op">)</span>,
                   n_probes <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="va">Name</span><span class="op">)</span><span class="op">)</span>,
                   contexts <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/paste.html" class="external-link">paste</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/unlist.html" class="external-link">unlist</a></span><span class="op">(</span><span class="fu">stringr</span><span class="fu">::</span><span class="fu"><a href="https://stringr.tidyverse.org/reference/str_split.html" class="external-link">str_split</a></span><span class="op">(</span><span class="va">UCSC_RefGene_Group</span>, <span class="st">";"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>, collapse <span class="op">=</span> <span class="st">";"</span><span class="op">)</span>,
                   .groups <span class="op">=</span> <span class="st">"drop"</span><span class="op">)</span>
<span class="va">genesEpic</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span> <span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="op">)</span>
<span class="co">#&gt; <span style="color: #949494;"># A tibble: 6 × 6</span></span>
<span class="co">#&gt;   chr   gene         from        to n_probes contexts                           </span>
<span class="co">#&gt;   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>       <span style="color: #949494; font-style: italic;">&lt;int&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;int&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>                              </span>
<span class="co">#&gt; <span style="color: #BCBCBC;">1</span> chr1  A3GALT2  33<span style="text-decoration: underline;">776</span>046  33<span style="text-decoration: underline;">787</span>381       12 Body;TSS200;TSS1500                </span>
<span class="co">#&gt; <span style="color: #BCBCBC;">2</span> chr1  AADACL3  12<span style="text-decoration: underline;">774</span>815  12<span style="text-decoration: underline;">787</span>380       12 Body;5'UTR;1stExon;TSS200;TSS1500;…</span>
<span class="co">#&gt; <span style="color: #BCBCBC;">3</span> chr1  AADACL4  12<span style="text-decoration: underline;">703</span>392  12<span style="text-decoration: underline;">727</span>093       17 TSS1500;Body;1stExon;TSS200;3'UTR  </span>
<span class="co">#&gt; <span style="color: #BCBCBC;">4</span> chr1  ABCA4    94<span style="text-decoration: underline;">458</span>645  94<span style="text-decoration: underline;">587</span>905       75 TSS1500;1stExon;5'UTR;Body;ExonBnd…</span>
<span class="co">#&gt; <span style="color: #BCBCBC;">5</span> chr1  ABCB10  229<span style="text-decoration: underline;">655</span>141 229<span style="text-decoration: underline;">695</span>511       27 TSS1500;TSS200;Body                </span>
<span class="co">#&gt; <span style="color: #BCBCBC;">6</span> chr1  ABCD3    94<span style="text-decoration: underline;">882</span>585  94<span style="text-decoration: underline;">983</span>186       24 TSS1500;1stExon;Body;TSS200;3'UTR;…</span></code></pre></div>
<p>How many probes are there per gene?</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://ggplot2.tidyverse.org" class="external-link">ggplot2</a></span><span class="op">)</span>
<span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="va">genesEpic</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">n_probes</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> 
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_histogram.html" class="external-link">geom_histogram</a></span><span class="op">(</span>binwidth <span class="op">=</span> <span class="fl">1</span><span class="op">)</span> <span class="op">+</span> 
  <span class="fu"><a href="../reference/intratumormeth_theme.html">intratumormeth_theme</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<p><img src="copynumber_files/figure-html/unnamed-chunk-5-1.png" width="700"></p>
<p>Keep only genes with at least 20 probes coverage:</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">minProbesPerGene</span> <span class="op">&lt;-</span> <span class="fl">20</span>

<span class="va">detailGenes</span> <span class="op">&lt;-</span> <span class="va">genesEpic</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span> 
  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">n_probes</span> <span class="op">&gt;=</span> <span class="va">minProbesPerGene</span><span class="op">)</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span> 
  <span class="co"># rename potentially duplicate names by adding the chromosome as a suffix</span>
  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span>gene <span class="op">=</span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/case_when.html" class="external-link">case_when</a></span><span class="op">(</span><span class="va">gene</span> <span class="op"><a href="https://rdrr.io/pkg/BiocGenerics/man/match.html" class="external-link">%in%</a></span> <span class="va">.</span><span class="op">$</span><span class="va">gene</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/duplicated.html" class="external-link">duplicated</a></span><span class="op">(</span><span class="va">.</span><span class="op">$</span><span class="va">gene</span><span class="op">)</span><span class="op">]</span> <span class="op">~</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="va">gene</span>, <span class="st">"_"</span>, <span class="va">chr</span><span class="op">)</span>,
                          <span class="cn">TRUE</span> <span class="op">~</span> <span class="va">gene</span><span class="op">)</span><span class="op">)</span>

<span class="va">detailRegions</span> <span class="op">&lt;-</span> <span class="fu">GenomicRanges</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/GRanges-class.html" class="external-link">GRanges</a></span><span class="op">(</span>seqnames <span class="op">=</span> <span class="va">detailGenes</span><span class="op">$</span><span class="va">chr</span>,
                         ranges <span class="op">=</span> <span class="fu">IRanges</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/IRanges/man/IRanges-constructor.html" class="external-link">IRanges</a></span><span class="op">(</span>start <span class="op">=</span> <span class="va">detailGenes</span><span class="op">$</span><span class="va">from</span>, end <span class="op">=</span> <span class="va">detailGenes</span><span class="op">$</span><span class="va">to</span><span class="op">)</span>,
                         name <span class="op">=</span> <span class="va">detailGenes</span><span class="op">$</span><span class="va">gene</span>,
                         <span class="co"># thick specifies the plotting window for detail plots</span>
                         thick <span class="op">=</span> <span class="fu">IRanges</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/IRanges/man/IRanges-constructor.html" class="external-link">IRanges</a></span><span class="op">(</span>start <span class="op">=</span> <span class="va">detailGenes</span><span class="op">$</span><span class="va">from</span> <span class="op">-</span> <span class="fl">5000</span>, end <span class="op">=</span> <span class="va">detailGenes</span><span class="op">$</span><span class="va">to</span> <span class="op">+</span> <span class="fl">5000</span><span class="op">)</span>,
                         seqinfo <span class="op">=</span> <span class="fu">GenomeInfoDb</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/GenomeInfoDb/man/Seqinfo-class.html" class="external-link">Seqinfo</a></span><span class="op">(</span>genome <span class="op">=</span> <span class="st">"hg19"</span><span class="op">)</span><span class="op">)</span>
<span class="va">detailRegions</span>
<span class="co">#&gt; GRanges object with 12724 ranges and 2 metadata columns:</span>
<span class="co">#&gt;           seqnames              ranges strand |        name               thick</span>
<span class="co">#&gt;              &lt;Rle&gt;           &lt;IRanges&gt;  &lt;Rle&gt; | &lt;character&gt;           &lt;IRanges&gt;</span>
<span class="co">#&gt;       [1]     chr1   94458645-94587905      * |       ABCA4   94453645-94592905</span>
<span class="co">#&gt;       [2]     chr1 229655141-229695511      * |      ABCB10 229650141-229700511</span>
<span class="co">#&gt;       [3]     chr1   94882585-94983186      * |       ABCD3   94877585-94988186</span>
<span class="co">#&gt;       [4]     chr1 179069967-179200312      * |        ABL2 179064967-179205312</span>
<span class="co">#&gt;       [5]     chr1   76188832-76215992      * |       ACADM   76183832-76220992</span>
<span class="co">#&gt;       ...      ...                 ...    ... .         ...                 ...</span>
<span class="co">#&gt;   [12720]     chrX 134419896-134479169      * |      ZNF75D 134414896-134484169</span>
<span class="co">#&gt;   [12721]     chrY   16634027-16945913      * |      NLGN4Y   16629027-16950913</span>
<span class="co">#&gt;   [12722]     chrY     9194502-9366496      * |       TSPY4     9189502-9371496</span>
<span class="co">#&gt;   [12723]     chrY   21154291-21240028      * |      TTTY14   21149291-21245028</span>
<span class="co">#&gt;   [12724]     chrY     2802103-2844835      * |         ZFY     2797103-2849835</span>
<span class="co">#&gt;   -------</span>
<span class="co">#&gt;   seqinfo: 298 sequences (2 circular) from hg19 genome</span></code></pre></div>
<p>Overall, 12,724 genetic loci are included in this detail regions object.</p>
<p>Check if all regions from Martin Sill’s annotation object are also contained:</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># this object has name 'anno'</span>
<span class="fu"><a href="https://rdrr.io/r/base/load.html" class="external-link">load</a></span><span class="op">(</span><span class="st">"~/data/cnv_ref/CNanalysis4_conumee_ANNO.vh20150715.RData"</span><span class="op">)</span>

<span class="co"># show detail region names not contained in my detail regions</span>
<span class="va">anno</span><span class="op">@</span><span class="va">detail</span><span class="op">[</span><span class="op">!</span><span class="va">anno</span><span class="op">@</span><span class="va">detail</span><span class="op">$</span><span class="va">name</span> <span class="op"><a href="https://rdrr.io/pkg/BiocGenerics/man/match.html" class="external-link">%in%</a></span> <span class="va">detailRegions</span><span class="op">$</span><span class="va">name</span>, <span class="op">]</span>
<span class="co">#&gt; GRanges object with 5 ranges and 2 metadata columns:</span>
<span class="co">#&gt;       seqnames              ranges strand |          name               thick</span>
<span class="co">#&gt;          &lt;Rle&gt;           &lt;IRanges&gt;  &lt;Rle&gt; |   &lt;character&gt;           &lt;IRanges&gt;</span>
<span class="co">#&gt;   [1]     chr4     1739326-1808410      * |   FGFR3/TACC3     1239326-2308410</span>
<span class="co">#&gt;   [2]     chr7 138552722-140487384      * | KIAA1549/BRAF 138052722-140987384</span>
<span class="co">#&gt;   [3]     chr8   38279316-38688695      * |   FGFR1/TACC1   37779316-39188695</span>
<span class="co">#&gt;   [4]     chr9   21967751-22014312      * |      CDKN2A/B   21467751-22514312</span>
<span class="co">#&gt;   [5]    chr19   54164928-54265684      * |         C19MC   53664928-54765684</span>
<span class="co">#&gt;   -------</span>
<span class="co">#&gt;   seqinfo: 22 sequences from an unspecified genome</span>

<span class="co"># add those genetic loci to my detail regions</span>
<span class="va">detailRegions</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="va">detailRegions</span>, <span class="va">anno</span><span class="op">@</span><span class="va">detail</span><span class="op">[</span><span class="op">!</span><span class="va">anno</span><span class="op">@</span><span class="va">detail</span><span class="op">$</span><span class="va">name</span> <span class="op"><a href="https://rdrr.io/pkg/BiocGenerics/man/match.html" class="external-link">%in%</a></span> <span class="va">detailRegions</span><span class="op">$</span><span class="va">name</span>, <span class="op">]</span><span class="op">)</span>
<span class="co"># check:</span>
<span class="va">anno</span><span class="op">@</span><span class="va">detail</span><span class="op">[</span><span class="op">!</span><span class="va">anno</span><span class="op">@</span><span class="va">detail</span><span class="op">$</span><span class="va">name</span> <span class="op"><a href="https://rdrr.io/pkg/BiocGenerics/man/match.html" class="external-link">%in%</a></span> <span class="va">detailRegions</span><span class="op">$</span><span class="va">name</span>, <span class="op">]</span>
<span class="co">#&gt; GRanges object with 0 ranges and 2 metadata columns:</span>
<span class="co">#&gt;    seqnames    ranges strand |        name     thick</span>
<span class="co">#&gt;       &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;character&gt; &lt;IRanges&gt;</span>
<span class="co">#&gt;   -------</span>
<span class="co">#&gt;   seqinfo: 22 sequences from an unspecified genome</span></code></pre></div>
<p>Five genomic regions were not contained, these were added manually.</p>
<p>Store this GRanges object for later use:</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/files2.html" class="external-link">dir.create</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="va">outputDir</span>, <span class="st">"copynumber"</span><span class="op">)</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/readRDS.html" class="external-link">saveRDS</a></span><span class="op">(</span><span class="va">detailRegions</span>, <span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="va">outputDir</span>, <span class="st">"copynumber"</span>, <span class="st">"cnv_detailregions_granges.Rdata"</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<p>Finally, the conumee annotation object can be generated taking our previously created detail regions object as input. Two versions are computed, one with and the other without sex chromosomes.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">detailRegionsNoXy</span> <span class="op">&lt;-</span> <span class="va">detailRegions</span><span class="op">[</span><span class="op">!</span><span class="op">(</span><span class="fu">seqnames</span><span class="op">(</span><span class="va">detailRegions</span><span class="op">)</span> <span class="op"><a href="https://rdrr.io/pkg/BiocGenerics/man/match.html" class="external-link">%in%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"chrX"</span>, <span class="st">"chrY"</span><span class="op">)</span><span class="op">)</span><span class="op">]</span>

<span class="va">cnvAnnoEpicXY</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/conumee/man/CNV.create_anno.html" class="external-link">CNV.create_anno</a></span><span class="op">(</span>array_type <span class="op">=</span> <span class="st">"EPIC"</span>, chrXY <span class="op">=</span> <span class="cn">TRUE</span>, detail_regions <span class="op">=</span> <span class="va">detailRegions</span><span class="op">)</span>
<span class="va">cnvAnnoEpic</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/conumee/man/CNV.create_anno.html" class="external-link">CNV.create_anno</a></span><span class="op">(</span>array_type <span class="op">=</span> <span class="st">"EPIC"</span>, chrXY <span class="op">=</span> <span class="cn">FALSE</span>, detail_regions <span class="op">=</span> <span class="va">detailRegionsNoXy</span><span class="op">)</span>

<span class="va">cnvAnnoEpic</span>
<span class="co">#&gt; CNV annotation object</span>
<span class="co">#&gt;    created  : Mon Mar 28 22:43:22 2022</span>
<span class="co">#&gt;   @genome   : 22 chromosomes</span>
<span class="co">#&gt;   @gap      : 313 regions</span>
<span class="co">#&gt;   @probes   : 844316 probes</span>
<span class="co">#&gt;   @exclude  : 0 regions (overlapping 0 probes)</span>
<span class="co">#&gt;   @detail   : 12443 regions (overlapping 639803 probes)</span>
<span class="co">#&gt;   @bins     : 25752 bins (min/avg/max size: 50/103.8/5000kb, probes: 15/32.8/437)</span></code></pre></div>
<p>Save them:</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/readRDS.html" class="external-link">saveRDS</a></span><span class="op">(</span><span class="va">cnvAnnoEpic</span>, <span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="va">outputDir</span>, <span class="st">"copynumber"</span>, <span class="st">"conumee_annotation_epic.Rdata"</span><span class="op">)</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/readRDS.html" class="external-link">saveRDS</a></span><span class="op">(</span><span class="va">cnvAnnoEpicXY</span>, <span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="va">outputDir</span>, <span class="st">"copynumber"</span>, <span class="st">"conumee_annotation_epic_xy.Rdata"</span><span class="op">)</span><span class="op">)</span></code></pre></div>
</div>
<div class="section level2">
<h2 id="run-conumee">Run conumee<a class="anchor" aria-label="anchor" href="#run-conumee"></a>
</h2>
<p>Load the annotation data created above:</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">annoF</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/readRDS.html" class="external-link">readRDS</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="va">inputDir</span>, <span class="st">"conumee"</span>, <span class="st">"conumee_annotation_epic.Rdata"</span><span class="op">)</span><span class="op">)</span>
<span class="va">annoM</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/readRDS.html" class="external-link">readRDS</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="va">inputDir</span>, <span class="st">"conumee"</span>, <span class="st">"conumee_annotation_epic_xy.Rdata"</span><span class="op">)</span><span class="op">)</span>
<span class="va">annoF</span>
<span class="co">#&gt; CNV annotation object</span>
<span class="co">#&gt;    created  : Mon Nov 29 11:44:40 2021</span>
<span class="co">#&gt;   @genome   : 22 chromosomes</span>
<span class="co">#&gt;   @gap      : 313 regions</span>
<span class="co">#&gt;   @probes   : 844316 probes</span>
<span class="co">#&gt;   @exclude  : 0 regions (overlapping 0 probes)</span>
<span class="co">#&gt;   @detail   : 12443 regions (overlapping 639803 probes)</span>
<span class="co">#&gt;   @bins     : 25752 bins (min/avg/max size: 50/103.8/5000kb, probes: 15/32.8/437)</span></code></pre></div>
<p>Load the reference data required by conumee:</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">refF</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/readRDS.html" class="external-link">readRDS</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="va">inputDir</span>, <span class="st">"conumee"</span>, <span class="st">"conumee_ref_epic_F_Sill.Rdata"</span><span class="op">)</span><span class="op">)</span>
<span class="va">refM</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/readRDS.html" class="external-link">readRDS</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="va">inputDir</span>, <span class="st">"conumee"</span>, <span class="st">"conumee_ref_epic_M_Sill.Rdata"</span><span class="op">)</span><span class="op">)</span>
<span class="va">refM</span>
<span class="co">#&gt; CNV data object</span>
<span class="co">#&gt;    created   : </span>
<span class="co">#&gt;   @intensity : available (50 samples, 865859 probes)</span></code></pre></div>
<p>Now all inputs are prepared and we can run conumee:</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/run_conumee.html">run_conumee</a></span><span class="op">(</span><span class="va">outputDir</span>, <span class="va">samplesheet</span>, <span class="va">patientsheet</span>, <span class="va">refF</span>, <span class="va">refM</span>, <span class="va">annoF</span>, <span class="va">annoM</span><span class="op">)</span></code></pre></div>
<p>This writes a <code>CNV.analysis</code> object for each sample into the specified output directory. Since these objects contain a lot of data I will extract the important information to avoid having to load large objects into memory later on.</p>
</div>
<div class="section level2">
<h2 id="genomeplots">Genomeplots<a class="anchor" aria-label="anchor" href="#genomeplots"></a>
</h2>
<p>Depending on how many detail regions were specified in the conumee annotation (around 12000 in this case), the resulting plot can become unreadable since each detail region name is printed as a label. Therefore, it is possible to manually specify the detail regions that should be printed onto the plot:</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">detailRegions</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"MDM4"</span>, <span class="st">"MYCN"</span>, <span class="st">"TET3"</span>, <span class="st">"GLI2"</span>, <span class="st">"IDH1"</span>, <span class="st">"EPHA3"</span>, <span class="st">"PIK3CA"</span>, 
<span class="st">"FGFR3/TACC3"</span>, <span class="st">"PDGFRA"</span>, <span class="st">"TET2"</span>, <span class="st">"TERT"</span>, <span class="st">"VEGFA"</span>, <span class="st">"MYB"</span>, <span class="st">"EGFR"</span>, 
<span class="st">"HGF"</span>, <span class="st">"CDK6"</span>, <span class="st">"MET"</span>, <span class="st">"KIAA1549/BRAF"</span>, <span class="st">"FGFR1/TACC1"</span>, <span class="st">"MYBL1"</span>, 
<span class="st">"MYC"</span>, <span class="st">"CDKN2A/B"</span>, <span class="st">"PTCH1"</span>, <span class="st">"TET1"</span>, <span class="st">"PTEN"</span>, <span class="st">"MGMT"</span>, <span class="st">"VEGFB"</span>, 
<span class="st">"CCND1"</span>, <span class="st">"CCND2"</span>, <span class="st">"CDK4"</span>, <span class="st">"MDM2"</span>, <span class="st">"RB1"</span>, <span class="st">"TP53"</span>, <span class="st">"NF1"</span>, <span class="st">"PPM1D"</span>, 
<span class="st">"C19MC"</span>, <span class="st">"SMARCB1"</span>, <span class="st">"NF2"</span><span class="op">)</span></code></pre></div>
<p>Generate a CNV genomeplot for every sample organized by patient:</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/cnv_genomeplots_per_patient.html">cnv_genomeplots_per_patient</a></span><span class="op">(</span><span class="va">outputDir</span>, <span class="va">samplesheet</span>, detailRegions <span class="op">=</span> <span class="va">detailRegions</span><span class="op">)</span></code></pre></div>
</div>
<div class="section level2">
<h2 id="cnv-post-processing">CNV Post-processing<a class="anchor" aria-label="anchor" href="#cnv-post-processing"></a>
</h2>
<ul>
<li>extract detail region CNV</li>
<li>extract segments
<ul>
<li>homogenize segments</li>
</ul>
</li>
<li>TODO: detect focal events where a detail region copy number is significantly different from its neighborhood</li>
</ul>
<p>The above <code><a href="../reference/run_conumee.html">run_conumee()</a></code> call generated a lot of data: 4.7 GB in this case. The is mostly because there is a lot of redundancy in the information that is stored for each sample. To accelerate further analysis and to avoid having to load 4.7 GB into memory, the function <code><a href="../reference/postprocess_conumee_files.html">postprocess_conumee_files()</a></code> extracts the essential information and saves it to the output directory.</p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/postprocess_conumee_files.html">postprocess_conumee_files</a></span><span class="op">(</span><span class="va">outputDir</span><span class="op">)</span></code></pre></div>
<p>The generated data can now be read in:</p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">segmentCnv</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/readRDS.html" class="external-link">readRDS</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="va">outputDir</span>, <span class="st">"copynumber"</span>, <span class="st">"cnv_segments.Rdata"</span><span class="op">)</span><span class="op">)</span>
<span class="va">segmentCnv</span>
<span class="co">#&gt; <span style="color: #949494;"># A tibble: 16,316 × 8</span></span>
<span class="co">#&gt;    sample_id      chr  start    end log2ratio_mean log2ratio_sd log2ratio_median</span>
<span class="co">#&gt;    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>        <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;int&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;int&gt;</span>          <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>        <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>            <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span></span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 1</span> GU-HGG-154_1     1 9.37<span style="color: #949494;">e</span>4 7.75<span style="color: #949494;">e</span>5        -<span style="color: #BB0000;">0.061</span>        0.115           -<span style="color: #BB0000;">0.064</span><span style="color: #BB0000; text-decoration: underline;">7</span></span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 2</span> GU-HGG-154_1     1 8.25<span style="color: #949494;">e</span>5 1.11<span style="color: #949494;">e</span>7        -<span style="color: #BB0000;">0.372</span>        0.078<span style="text-decoration: underline;">8</span>          -<span style="color: #BB0000;">0.366</span> </span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 3</span> GU-HGG-154_1     1 1.12<span style="color: #949494;">e</span>7 1.54<span style="color: #949494;">e</span>7        -<span style="color: #BB0000;">0.040</span><span style="color: #BB0000; text-decoration: underline;">2</span>       0.060<span style="text-decoration: underline;">8</span>          -<span style="color: #BB0000;">0.046</span><span style="color: #BB0000; text-decoration: underline;">9</span></span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 4</span> GU-HGG-154_1     1 1.55<span style="color: #949494;">e</span>7 1.61<span style="color: #949494;">e</span>7         0.160        0.028<span style="text-decoration: underline;">9</span>           0.159 </span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 5</span> GU-HGG-154_1     1 1.61<span style="color: #949494;">e</span>7 2.72<span style="color: #949494;">e</span>7        -<span style="color: #BB0000;">0.036</span><span style="color: #BB0000; text-decoration: underline;">1</span>       0.052<span style="text-decoration: underline;">7</span>          -<span style="color: #BB0000;">0.033</span><span style="color: #BB0000; text-decoration: underline;">2</span></span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 6</span> GU-HGG-154_1     1 2.72<span style="color: #949494;">e</span>7 3.28<span style="color: #949494;">e</span>7        -<span style="color: #BB0000;">0.393</span>        0.066<span style="text-decoration: underline;">4</span>          -<span style="color: #BB0000;">0.389</span> </span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 7</span> GU-HGG-154_1     1 3.29<span style="color: #949494;">e</span>7 1.43<span style="color: #949494;">e</span>8        -<span style="color: #BB0000;">0.021</span><span style="color: #BB0000; text-decoration: underline;">7</span>       0.047<span style="text-decoration: underline;">2</span>          -<span style="color: #BB0000;">0.021</span><span style="color: #BB0000; text-decoration: underline;">8</span></span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 8</span> GU-HGG-154_1     1 1.44<span style="color: #949494;">e</span>8 1.45<span style="color: #949494;">e</span>8         0.128        0.062<span style="text-decoration: underline;">7</span>           0.119 </span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 9</span> GU-HGG-154_1     1 1.45<span style="color: #949494;">e</span>8 1.52<span style="color: #949494;">e</span>8         0.261        0.077            0.260 </span>
<span class="co">#&gt; <span style="color: #BCBCBC;">10</span> GU-HGG-154_1     1 1.52<span style="color: #949494;">e</span>8 1.53<span style="color: #949494;">e</span>8         0.397        0.072<span style="text-decoration: underline;">4</span>           0.407 </span>
<span class="co">#&gt; <span style="color: #949494;"># … with 16,306 more rows, and 1 more variable: log2ratio_mad &lt;dbl&gt;</span></span>
<span class="va">detailCnv</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/readRDS.html" class="external-link">readRDS</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="va">outputDir</span>, <span class="st">"copynumber"</span>, <span class="st">"cnv_detailregions.Rdata"</span><span class="op">)</span><span class="op">)</span>
<span class="va">detailCnv</span>
<span class="co">#&gt; <span style="color: #949494;"># A tibble: 3,273,218 × 6</span></span>
<span class="co">#&gt;    sample_id    region   log2ratio_median   chr   start     end</span>
<span class="co">#&gt;    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>        <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>               <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;int&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;int&gt;</span></span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 1</span> GU-HGG-154_1 SAMD11             -<span style="color: #BB0000;">0.326</span>     1  <span style="text-decoration: underline;">859</span>719  <span style="text-decoration: underline;">879</span>958</span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 2</span> GU-HGG-154_1 NOC2L              -<span style="color: #BB0000;">0.377</span>     1  <span style="text-decoration: underline;">879</span>958  <span style="text-decoration: underline;">895</span>988</span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 3</span> GU-HGG-154_1 KLHL17             -<span style="color: #BB0000;">0.338</span>     1  <span style="text-decoration: underline;">894</span>587  <span style="text-decoration: underline;">901</span>062</span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 4</span> GU-HGG-154_1 PLEKHN1            -<span style="color: #BB0000;">0.312</span>     1  <span style="text-decoration: underline;">900</span>550  <span style="text-decoration: underline;">910</span>223</span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 5</span> GU-HGG-154_1 AGRN               -<span style="color: #BB0000;">0.287</span>     1  <span style="text-decoration: underline;">954</span>619  <span style="text-decoration: underline;">990</span>557</span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 6</span> GU-HGG-154_1 C1orf159           -<span style="color: #BB0000;">0.306</span>     1 1<span style="text-decoration: underline;">017</span>383 1<span style="text-decoration: underline;">053</span>027</span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 7</span> GU-HGG-154_1 TTLL10             -<span style="color: #BB0000;">0.270</span>     1 1<span style="text-decoration: underline;">108</span>521 1<span style="text-decoration: underline;">133</span>255</span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 8</span> GU-HGG-154_1 SDF4               -<span style="color: #BB0000;">0.288</span>     1 1<span style="text-decoration: underline;">152</span>547 1<span style="text-decoration: underline;">168</span>550</span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 9</span> GU-HGG-154_1 FAM132A            -<span style="color: #BB0000;">0.274</span>     1 1<span style="text-decoration: underline;">177</span>886 1<span style="text-decoration: underline;">183</span>528</span>
<span class="co">#&gt; <span style="color: #BCBCBC;">10</span> GU-HGG-154_1 UBE2J2             -<span style="color: #BB0000;">0.432</span>     1 1<span style="text-decoration: underline;">189</span>466 1<span style="text-decoration: underline;">210</span>611</span>
<span class="co">#&gt; <span style="color: #949494;"># … with 3,273,208 more rows</span></span></code></pre></div>
</div>
<div class="section level2 unnumbered">
<h2 class="unnumbered" id="references">References<a class="anchor" aria-label="anchor" href="#references"></a>
</h2>
<div id="refs" class="references csl-bib-body hanging-indent">
<div id="ref-hovestadt_conumee_2017" class="csl-entry">
Hovestadt, Volker, and Marc Zapatka. 2017. <span>“Conumee: Enhanced Copy-Number Variation Analysis Using Illumina <span>DNA</span> Methylation Arrays.”</span>
</div>
</div>
</div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p></p>
<p>Developed by Fynn Withake.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.2.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
