<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="intratumormeth">
<title>Preprocessing • intratumormeth</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.1.0/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.1.0/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.rawgit.com/afeld/bootstrap-toc/v1.0.1/dist/bootstrap-toc.min.js"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Preprocessing">
<meta property="og:description" content="intratumormeth">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">intratumormeth</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.0.0.9000</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="active nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <h6 class="dropdown-header" data-toc-skip>Analysis workflow</h6>
    <a class="dropdown-item" href="../articles/preprocessing.html">Preprocessing</a>
    <a class="dropdown-item" href="../articles/quality_control.html">Quality Control</a>
    <a class="dropdown-item" href="../articles/copynumber.html">Copynumber</a>
  </div>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav"></ul>
</div>

    
  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main"><div class="page-header">
      <img src="" class="logo" alt=""><h1>Preprocessing</h1>
            
      
      
      <div class="d-none name"><code>preprocessing.Rmd</code></div>
    </div>

    
    
<div class="section level2">
<h2 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<p>This vignette shows the preprocessing workflow used to analyze raw IDAT files from Illumina HumanMethylationEPIC microarrays. After reading in the intensity levels, the following steps will be performed:</p>
<ul>
<li>calculate detection p-values based on control probes using <code><a href="https://rdrr.io/pkg/minfi/man/detectionP.html" class="external-link">minfi::detectionP()</a></code>,</li>
<li>conversion of intensity levels into methylation beta-values and M-values using <code><a href="https://rdrr.io/pkg/minfi/man/preprocessRaw.html" class="external-link">minfi::preprocessRaw()</a></code>,</li>
<li>normalization in order to account for the two different probe types employed on the array and other systematic biases of the platform via:
<ul>
<li>
<code><a href="https://rdrr.io/pkg/minfi/man/preprocessSwan.html" class="external-link">minfi::preprocessSWAN()</a></code>,</li>
<li>
<code><a href="https://rdrr.io/pkg/minfi/man/preprocessNoob.html" class="external-link">minfi::preprocessNoob()</a></code>,</li>
<li>
<code><a href="https://rdrr.io/pkg/minfi/man/preprocessQuantile.html" class="external-link">minfi::preprocessQuantile()</a></code>.</li>
</ul>
</li>
</ul>
<p>The resulting data will be saved to the specified output directory.</p>
<p>After this first preprocessing step, methylation data can be obtained using function <code>get_beta()</code> which allows to specify the normalization method, as well as a probe filter and a sample filter. Probe filtering has been suggested repeatedly in literature since some array probes were found to yield non-reproducible results due to targeting SNPs instead of methylation, non-unique binding sequences or off-target hybridization <span class="citation">(Zhou, Laird, and Shen 2016; Pidsley et al. 2016)</span>.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">intratumormeth</span><span class="op">)</span></code></pre></div>
</div>
<div class="section level2">
<h2 id="data-format-requirements">Data format requirements<a class="anchor" aria-label="anchor" href="#data-format-requirements"></a>
</h2>
<p>To run the preprocessing workflow, a <code>samplesheet</code> dataframe has to be prepared which associates a given sample id with the location of corresponding IDAT files. There should be one green and one red channel IDAT file for each sample which can be distinguished by their file name suffix (e.g. <code>filename_Red.idat</code> and <code>filename_Grn.idat</code>). Column <code>idat_location</code> should contain the filename <em>without</em> the red/green suffix. Since this is an <em>intratumor</em> heterogeneity study, the individual samples furthermore have to be associated with corresponding patients. The patient id corresponding to a given sample should be specified in column <code>patient_id</code>. Additionally, arbitrary sample-wise metadata can be specified by other columns in the samplesheet (take care not to name them ambiguously).</p>
<p>An exemplary samplesheet is shown here:</p>
<pre><code><span class="co">#&gt;   sample_id patient_id                                       idat_location</span>
<span class="co">#&gt; 1   sfb01_1      sfb01 ~/data/methylation_data/Pat01.1/202212330261_R07C01</span>
<span class="co">#&gt; 2   sfb01_2      sfb01 ~/data/methylation_data/Pat01.2/202242420132_R07C01</span>
<span class="co">#&gt; 3   sfb01_3      sfb01 ~/data/methylation_data/Pat01.3/202212330261_R06C01</span>
<span class="co">#&gt; 4   sfb02_1      sfb02 ~/data/methylation_data/Pat02.1/202242420132_R02C01</span>
<span class="co">#&gt; 5   sfb02_2      sfb02 ~/data/methylation_data/Pat02.2/202242420132_R06C01</span>
<span class="co">#&gt; 6   sfb02_3      sfb02 ~/data/methylation_data/Pat02.3/202242420132_R01C01</span>
<span class="co">#&gt;   project</span>
<span class="co">#&gt; 1  SFB824</span>
<span class="co">#&gt; 2  SFB824</span>
<span class="co">#&gt; 3  SFB824</span>
<span class="co">#&gt; 4  SFB824</span>
<span class="co">#&gt; 5  SFB824</span>
<span class="co">#&gt; 6  SFB824</span></code></pre>
<p>This samplesheet contains 213 samples from 46 patients of the SFB824 project B12, as well as 47 samples from 15 patients from the dataset published by <span class="citation">Wenger et al. (2019)</span> (GEO accession: GSE116298)</p>
</div>
<div class="section level2">
<h2 id="workflow">Workflow<a class="anchor" aria-label="anchor" href="#workflow"></a>
</h2>
<p>Now all required data has been prepared and the actual preprocessing can begin. To be able to render this vignette faster, only a 10 sample subset of the data will be processed.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/preprocess_methylation_data.html">preprocess_methylation_data</a></span><span class="op">(</span><span class="va">samplesheet</span>, outputDir <span class="op">=</span> <span class="st">"~/sfb824/packagepdgfra_output/"</span><span class="op">)</span></code></pre></div>
<p>Once finished, the generated data can be obtained like so:</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">betas</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/get_betas.html">get_betas</a></span><span class="op">(</span>outputDir <span class="op">=</span> <span class="st">"~/sfb824/packagepdgfra_output/"</span>, 
                   normalization <span class="op">=</span> <span class="st">"swan"</span>, 
                   removeNaProbes <span class="op">=</span> <span class="cn">TRUE</span>,
                   probeFilter <span class="op">=</span> <span class="cn">NULL</span>,
                   sampleFilter <span class="op">=</span> <span class="cn">NULL</span><span class="op">)</span>
<span class="va">betas</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">5</span>,<span class="fl">1</span><span class="op">:</span><span class="fl">5</span><span class="op">]</span></code></pre></div>
<p>Before starting further analysis of these methylation levels, a quality control step should be performed in order to detect samples of poor quality (i.e. high average detection p-value) and filter out probes that are associated with low reproducibility <span class="citation">(Zhou, Laird, and Shen 2016; Pidsley et al. 2016)</span>. See vignette <code>quality_control</code> for further details.</p>
</div>
<div class="section level2 unnumbered">
<h2 class="unnumbered" id="references">References<a class="anchor" aria-label="anchor" href="#references"></a>
</h2>
<div id="refs" class="references csl-bib-body hanging-indent">
<div id="ref-pidsley_critical_2016" class="csl-entry">
Pidsley, Ruth, Elena Zotenko, Timothy J. Peters, Mitchell G. Lawrence, Gail P. Risbridger, Peter Molloy, Susan Van Djik, Beverly Muhlhausler, Clare Stirzaker, and Susan J. Clark. 2016. <span>“Critical Evaluation of the Illumina <span>MethylationEPIC</span> <span>BeadChip</span> Microarray for Whole-Genome <span>DNA</span> Methylation Profiling.”</span> <em>Genome Biology</em> 17 (1): 208. <a href="https://doi.org/10.1186/s13059-016-1066-1" class="external-link">https://doi.org/10.1186/s13059-016-1066-1</a>.
</div>
<div id="ref-wenger_intratumor_2019" class="csl-entry">
Wenger, Anna, Sandra Ferreyra Vega, Teresia Kling, Thomas Olsson Bontell, Asgeir Store Jakola, and Helena Carén. 2019. <span>“Intratumor <span>DNA</span> Methylation Heterogeneity in Glioblastoma: Implications for <span>DNA</span> Methylation-Based Classification.”</span> <em>Neuro-Oncology</em> 21 (5): 616–27. <a href="https://doi.org/10.1093/neuonc/noz011" class="external-link">https://doi.org/10.1093/neuonc/noz011</a>.
</div>
<div id="ref-zhou_comprehensive_2016" class="csl-entry">
Zhou, Wanding, Peter W. Laird, and Hui Shen. 2016. <span>“Comprehensive Characterization, Annotation and Innovative Use of Infinium <span>DNA</span> Methylation <span>BeadChip</span> Probes.”</span> <em>Nucleic Acids Research</em>, October, gkw967. <a href="https://doi.org/10.1093/nar/gkw967" class="external-link">https://doi.org/10.1093/nar/gkw967</a>.
</div>
</div>
</div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p></p>
<p>Developed by Fynn Withake.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.2.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
